docs: add tailwindcss for styles, oidc authorize endpoint playground (#4707)
zitadel/zitadel: docs/docs/apis/openidoauth/authrequest.mdx

---
title: OIDC Authentication Request Playground
sidebar_label: OIDC Playground
---

import { SetAuthRequest } from "../../../src/components/authrequest";

The OIDC Playground is for testing OpenID Authentication Requests, giving you more insight how OpenID Connect works and how you can customize ZITADEL behavior with different parameters.

> An Open ID Connect (OIDC) [authentication request](https://openid.net/specs/openid-connect-core-1_0.html) is an OAuth 2.0 Authorization Request using additional parameters and scopes to request that the end-user be authenticated by ZITADEL.

Reviewer: ```suggestion
An OpenID Connect (OIDC) [authentication request](https://openid.net/specs/openid-connect-core-1_0.html) is an OAuth 2.0 Authorization Request using additional parameters and scopes to request that the end-user be authenticated by ZITADEL.
```

<SetAuthRequest />

## Why this OIDC playground?

> In this version of ZITADEL, human end-users need to authenticate trough a hosted login page.
> Your application should initiate a login by issuing an authentication request and redirecting the user to the login page. You can also customize the behavior of ZITADEL by providing additional parameters and scopes in the request.

Reviewer: ```suggestion
Currently ZITADEL requires human users to authenticate trough the hosted login page.
Your application should initiate a login by issuing an authentication request and redirecting the user to the login page. You can customize the behavior of ZITADEL by providing additional parameters and scopes in the request.
```

> This playground should help you to initialially craft an authentication request to explore the behavior of ZITADEL in more detail.

Reviewer: ```suggestion
This playground should help you to initially craft an authentication request and to explore the behavior of ZITADEL in more depth.
```

## Request parameters explained

Not all request parameters are available in the playground. Please refer to the full documentation of the [authorization endpoint](/docs/apis/openidoauth/endpoints#authorization_endpoint).

### Your Domain

The <span className="text-yellow-500">Instance Domain</span> to your ZITADEL instance. Use the base-path, the playground will add the required path to the request.

### Required Parameters

<p>
  <span className="text-green-500">Client ID</span> is the resource id of an
  application. It's the application where you want your users to login. You can
  find the resource id in the Console. When using organization grants, use the
  client id from the origin organization.
</p>

<p>
  <span className="text-blue-500">Redirect URI</span> be one of the
  pre-configured redirect uris for your application. You must add the redirect
  uri for your application, else you will receive an error.
</p>

<p>

>   <span className="text-orange-500">Response Type</span> termines whether a
>   code, id_token token or just id_token will be returned. Most use cases will
>   need code.

Reviewer: ```suggestion
  <span className="text-orange-500">Response Type</span> defines whether a
  code, id_token token or just id_token will be returned. Most use cases will
  need code.
```

## Standard Scopes

Not all scopes are available in the playground. Please refer to the full [documentation](/docs/apis/openidoauth/scopes) for the exhaustive list of available standard and reserved scopes.

Used to request additional information from ZITADEL.
These scopes are defined in the OpenID Connect specification.
The `openid` scope is mandatory.

## Reserved Scopes

You can request additional information that is specific to ZITADEL or customize the behavior of ZITADEL by including reserved scopes.
Please refer to the documentation for a full list of available reserved scopes.

### Organization policies and branding

Enforce a organization's policies and branding as well as membership of the user by passing the scope `urn:zitadel:iam:org:id:{id}` with the required <span className="text-purple-500">Organization ID</span>.

Please refer to the full [guide on user metadata](/docs/guides/manage/customize/branding).

### Get user metadata

Pass the scope `urn:zitadel:iam:user:metadata` to get a user's metadata.
Please refer to the full [guide on branding](/docs/guides/manage/customize/user-metadata) for further details.

### Access core apis

> Calling the [core API](/docs/apis/introduction) with the authenticated user, requires that the project ID is included in the audience claim.

Reviewer: ```suggestion
Calling the [core API](/docs/apis/introduction) with the authenticated user, requires that the projectID of ZITADEL is included in the audience claim.
```

This can be achieved by adding the scope `urn:zitadel:iam:org:project:id:zitadel:aud` to your applications authorization request.

> ## How to use in your project

Reviewer: ```suggestion
## How to use ZITADEL in your project
```
